### DISCUSSION TITLE
Could zod be TS first?

### DISCUSSION THREAD

**Discussion by adiled** [colinhacks/zod#1728] [General]
I saw a demo and it seems the workflow demands plastering Zs all over, unless of course you limit the use, even then it's Zs in that scope.

What is the feasibility of Zod as TS plugin to switch the dev style from inference to deference from TS structure? I feel a little dread thinking about this becoming a norm.

**Comment by adiled** [comment]
Alright, not possible: https://github.com/microsoft/TypeScript/issues/16607

**Comment by JacobWeisenburger** [comment]
It might help others to be able to following along with what you are saying, if you were to include some code examples that explain more about what you are describing. thanks

**Reply by adiled** [reply]
I meant typescript compiler code transformation plugin that would deduce zodding from types but `tsc` doesn't seem to have taken that direction. It's safe to close this discussion.
